File: solutions/EightRooks/solve/ruby/EightRooksSolver.rb
#! /usr/bin/env ruby

require_relative "EightRooks"

require "gettc/types"
include Gettc

def init()
    gettc_home = File.join(ENV["HOME"], ".gettc")
    if ENV.has_key?("GETTC_HOME")
        gettc_home = ENV["GETTC_HOME"]
    end 
    gettc_home = File.expand_path(gettc_home)
    $LOAD_PATH << File.join(gettc_home, "include", "ruby")
    require "topcoder"
    include TopCoder
end

def main()
    reader = Reader.new(IO.read(ARGV[0]))
    board = reader.next(TArray.new(TString))

    result = EightRooks.new().isCorrect(board)
    IO.write(ARGV[1], Writer.new().next(result, TString).to_s())
end
init()
main()

File: solutions/EightRooks/solve/java/EightRooksSolver.java
import java.io.FileReader;
import java.io.FileWriter;

import java.util.List;
import java.util.ArrayList;

import org.topcoder.TopcoderReader;
import org.topcoder.TopcoderWriter;
import org.topcoder.TypeRef;

public class EightRooksSolver {
    public static void main(String[] args) {
    try {
        TopcoderReader reader = new TopcoderReader(new FileReader(args[0]));
        List<String> boardBoxed = (List<String>) reader.next(new TypeRef<List<String>>(){}.getType());
        String[] board = new String[boardBoxed.size()];
        for (int _i = 0; _i < boardBoxed.size(); ++_i)
            board[_i] = boardBoxed.get(_i);
        reader.close();

        EightRooks solver = new EightRooks();
        TopcoderWriter writer = new TopcoderWriter(new FileWriter(args[1]));
        writer.write(solver.isCorrect(board));
        writer.close();
    } catch (Exception err) {
        err.printStackTrace(System.err);
    }
    }    
}
